analysis.py

```diff
@@ -110,9 +110,23 @@ def gpggaParse(self, gpgga, line):
         gpgga[6] = line[8]
         gpgga[7] = line[9] + line[10]
         gpgga[8] = line[11] + line[12]
-        gpgga[9] = line[13]
-        print(gpgga)
-        return gpgga
+        gpgga[9] = line[14]
+        
+        if self.verifyChecksum(line, gpgga[9]):
+            return gpgga
+        else:
+            raise Exception("Checksum is incorrect! \n" +
+                            "gpgga: " + str(gpgga) + "\n" +
+                            "line: " + str(line))
 
+    def verifyChecksum(self, line, checksum):
+        wholeString = ",".join(line)
+        calculatedSum = 0
+        for char in wholeString[1:-5]:
```

Comment:
The checksum is an XOR of all chars between $ and * in the sentence, so this takes everything but the $ and removes the * and everything after it.
Moved the [1:-5] to line 123 and commented it for clarity.